packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -167,14 +169,16 @@ export function PanelChrome({
     <>
       {/* Non collapsible title */}
       {!collapsible && title && (
-        <h6 title={typeof title === 'string' ? title : undefined} className={styles.title}>
-          {title}
-        </h6>
+        <div className={styles.title}>
+          <Text element="h2" variant="h6" truncate title={typeof title === 'string' ? title : undefined}>
+            {title}
+          </Text>
+        </div>
       )}
 
       {/* Collapsible title */}
       {collapsible && (
-        <h6 className={styles.title}>
+        <div className={styles.title}>
```

Comment:
The better solution here would probably to have the button next to the text, but still have the text clickable. Just having a button wrapping everything causes super confusing behaviour with a screen reader.

For now, this is a section with a label, so it is visible with the landmark.

Reply:
Still looking into making truncation work with button > h2. The screen reader seems to handle it well at least.